### PR TITLE
Added Azure VM sensor feature

### PR DESCRIPTION
**Pull request by vijayshinva**
**Description:**
The Azure VM Sensor displays the current status of a virtual machine hosted in the Microsoft Azure cloud platform. This will be useful to monitor servers like VPN/Email servers which HomeAssistant users use.

```yaml
# Example configuration.yaml entry

azure:
  client_id: 96b9XXXX-XXXX-XXXX-XXXX-XXXXcef0eff8
  client_secret: SECRET-PASSWORD
  tenant_id: 7912XXXX-XXXX-XXXX-XXXX-XXXX69bbace1
  subscription_id: f231XXXX-XXXX-XXXX-XXXX-XXXXa3299dea

sensor:
  - platform: azure_vm
    name: hassio-test-vm
    resource_group: hassio-test-resource-group
```
## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/azure.markdown [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+description: "Instructions on how to setup the Azure component."
+date: 2018-08-28 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: microsoft_azure.png
+ha_category: Others
+featured: true
+ha_release: 0.77.1
```
Comment (fabaff) [inline source/_components/azure.markdown]: New platform and components are only added to major releases. Thus, `0.78` or `0.79`.

**Review by DubhAd — DISMISSED** [review]
It'd be nice to encourage the use of secrets, but 👍

File: source/_components/azure.markdown [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+To integrate a Microsoft Azure Subscription with Home Assistant, add the following section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+azure:
+  client_id: 96b9XXXX-XXXX-XXXX-XXXX-XXXXcef0eff8
+  client_secret: SECRET-PASSWORD
+  tenant_id: 7912XXXX-XXXX-XXXX-XXXX-XXXX69bbace1
+  subscription_id: f231XXXX-XXXX-XXXX-XXXX-XXXXa3299dea
+
```
Comment (tjorim) [inline source/_components/azure.markdown]: Please remove this empty line since it wil show on the page

**Review by frenck — CHANGES_REQUESTED** [review]
@vijayshinva some small comments where raised. Could you take a look? 👍

File: source/_components/azure.markdown (hunk 2 above)
Comment (vijayshinva) [reply source/_components/azure.markdown]: done

**Comment by frenck** [comment]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Review by frenck — APPROVED** [review]

**Pull request closed without merging**